data-prepper-plugins/kafka-source/src/test/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceConfigTest.java

```diff
@@ -0,0 +1,80 @@
+package org.opensearch.dataprepper.plugins.kafka.source;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.opensearch.dataprepper.plugins.kafka.source.configuration.ConsumerConfigs;
+import org.opensearch.dataprepper.plugins.kafka.source.configuration.TopicsConfig;
+import org.yaml.snakeyaml.Yaml;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.opensearch.dataprepper.plugins.kafka.source.configuration.SchemaConfig;
+
+class KafkaSourceConfigTest {
+
+	@Mock
+	KafkaSourceConfig kafkaSourceConfig;
+
+	@Mock
+	Properties properties;
+
+	@Mock
+    ConsumerConfigs consumerConfigs;
+
+	@Mock
+	SchemaConfig schemaConfig;
+
+	List<String> bootstrapServers;
+
+	List<TopicsConfig> topicsConfigs;
+
+	@BeforeEach
+	void setUp() throws IOException {
+		kafkaSourceConfig = new KafkaSourceConfig();
+		consumerConfigs = new ConsumerConfigs();
+		schemaConfig = new SchemaConfig();
+		//Added to load Yaml file - Start
+		Yaml yaml = new Yaml();
+		FileReader reader = new FileReader(getClass().getClassLoader().getResource("sample-pipelines.yaml").getFile());
+		Object data = yaml.load(reader);
+		if(data instanceof Map){
+			Map<String, Object> propertyMap = (Map<String, Object>) data;
+			Map<String, Object> nestedMap = new HashMap<>();
+			iterateMap(propertyMap, nestedMap);
+			bootstrapServers = (List<String>) nestedMap.get("bootstrap_servers");
+			topicsConfigs = (List<TopicsConfig>) nestedMap.get("topics");
+			kafkaSourceConfig.setBootStrapServers(bootstrapServers);
```

Comment:
This test is loading a `Map` and then setting values. This is not really testing the behavior.

The way to test the configs is to use a Jackson `ObjectMapper`.

```
ObjectMapper objectMapper = new ObjectMapper();
KafkaSourceConfig actualConfig = objectMapper.readValue(reader, KafkaSourceConfig.class);

assertThat(actualConfig, notNullValue());
assertThat(actualConfig.getBootstrapServers(), notNullValue());
assertThat(actualConfig.getBootstrapServers().size(), equalTo(5));
assertThat(actualConfig.getBootstrapServers(), hasItem("127.0.0.1:9092"));
assertThat(actualConfig.getBootstrapServers(), hasItem("127.0.0.1:9093"));
// continue validating these...
assertThat(actualConfig.getTopics(), notNullValue());
// Same concept as above, just now for topics. Again see my other comments that this will change.
```